.hooks/check_dir_names.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+import sys
+import os
+import re
+
+SNAKE_CASE_REGEX = re.compile(r'^[\.a-z0-9_]+$')
+
+def is_snake_case(name):
+    return bool(SNAKE_CASE_REGEX.fullmatch(name))
+
+def main():
+    # Paths passed by pre-commit (filtered by exclude)
+    paths = sys.argv[1:]
+    #print(paths)
+    invalid_dirs = set()
+
+    for path in paths:
+        # Get all directory parts in the path
+        # Normalize path separators, skip leading './'
+        parts = os.path.normpath(path).split(os.sep)
+
+        # Ignore the filename (last part) if it's a file
+        if os.path.isfile(path) or ('.' in parts[-1]):
```

Comment:
oh we actually want snake case filenames too, not just directories

Reply:
So we can't have files like README.md together with parts of the source code?

Reply:
ah and tscn and tres and res and... any other godot things, if there are any others

anything someone could stick in a load() or preload()

but still, merge this as it is, it gets us 99% of the way there which is awesome 😊